modules/light-clients/08-wasm/types/vm.go

```diff
@@ -139,19 +144,72 @@ func wasmSudo[T ContractResult](ctx sdk.Context, clientStore storetypes.KVStore,
 	if err := json.Unmarshal(resp.Data, &result); err != nil {
 		return result, errorsmod.Wrap(ErrWasmInvalidResponseData, err.Error())
 	}
+
+	oldCodeHash := cs.CodeHash
+	newClientState, err := validatePostExecutionClientState(clientStore, cdc)
+	if err != nil {
+		return result, err
+	}
+
+	// code has should only be able to be modified during migration.
+	if !bytes.Equal(oldCodeHash, newClientState.CodeHash) {
+		return result, errorsmod.Wrapf(ErrWasmInvalidContractModification, "code hash changed from %s to %s", hex.EncodeToString(oldCodeHash), hex.EncodeToString(newClientState.CodeHash))
```

Comment:
```suggestion
		return result, errorsmod.Wrapf(ErrWasmInvalidContractModification, "expected code hash %s, got %s", hex.EncodeToString(oldCodeHash), hex.EncodeToString(newClientState.CodeHash))
```

Reply:
considering we modify `ClientState.codeHash` before calling the migrate endpoint (in `MigrateContract`) this check would be sensible in `wasmMigrate` too, no? 

I think `validatePostExecutionClientState` can include this check and we can unconditionally add it in both `wasmMigrate` and `wasminstantiate` (which could also write a janky code hash if it wanted).

Reply:
> considering we modify ClientState.codeHash before calling the migrate endpoint (in MigrateContract) this check would be sensible in wasmMigrate too, no?

But in `MigrateContract` we do allow the code hash to be changed, right?

Reply:
yup but it should only switch to the one we want to switch to, no?

Reply:
ah, I get what you mean now. But for `MigrateContract` [08-wasm sets the new code hash after calling the contract](https://github.com/cosmos/ibc-go/blob/243f36ff4fb6f967db2084d696af394d5fd5b95e/modules/light-clients/08-wasm/keeper/keeper.go#L172), so we have the control to set it right. Still, I see your point of checking that the contract doesn't change the code hash to alert that it is behaving unexpectedly.

Reply:
yea! I think its sensible to have this check for all entry points which have mutable access to store, will also unify reasoning about them all.